src/easyscience/base_classes/model_base.py

```
@@ -0,0 +1,115 @@
+
+    def get_fittable_parameters(self) -> List[Parameter]:
+        """
+        Get all parameters which can be fitted as a list.
+
+        :return: List of `Parameter` objects.
+        """
+        return [param for param in self.get_all_parameters() if param.independent]
+
+    def get_free_parameters(self) -> List[Parameter]:
+        """
+        Get all parameters which are currently free to be fitted as a list.
+
+        :return: List of `Parameter` objects.
+        """
+        return [param for param in self.get_fittable_parameters() if not param.fixed]
+
+    def get_fit_parameters(self) -> List[Parameter]:
+        """
+        This is an alias for `get_free_parameters`.
+        To be removed when fully moved to new base classes and minimizer can be changed.
+        """
+        return self.get_free_parameters()
+
+    @classmethod
+    def from_dict(cls, obj_dict: Dict[str, Any]) -> None:
+        """
+        Re-create an EasyScience object with DescriptorNumber attributes from a full encoded dictionary.
+
+        :param obj_dict: dictionary containing the serialized contents (from `SerializerDict`) of an EasyScience object
+        :return: Reformed EasyScience object
+        """
+        if not SerializerBase._is_serialized_easyscience_object(obj_dict):
+            raise ValueError('Input must be a dictionary representing an EasyScience object.')
+        if obj_dict['@class'] == cls.__name__:
+            kwargs = SerializerBase.deserialize_dict(obj_dict)
+            parameter_placeholder = {}
+            for key, value in kwargs.items():
+                if isinstance(value, DescriptorNumber):
+                    parameter_placeholder[key] = value
+                    kwargs[key] = value.value
+            cls_instance = cls(**kwargs)
+            for key, value in parameter_placeholder.items():
+                temp_param = getattr(cls_instance, key)
+                setattr(cls_instance, '_' + key, value)
```

`getattr` and `setattr` could fail. Consider wrapping them in try..except

---

They should only fail if we have implemented our classes incorrectly, in which case we should let them fail and throw an error to know about it, or if the serialized dict has been tampered with, in which case other pieces could fail anyways and we should also just let it fail.
Do you have a good reason as to why this could fail otherwise?

---

e.g. to let users know that the failure stems from this issue. Bubbled exceptions often lose context and it's difficult to pinpoint what happened, if you're in a library code

---

Is this (the new code) what you had in mind?